client/app/(auth)/forgotPassword.tsx

```diff
@@ -0,0 +1,167 @@
+import firestore from "@react-native-firebase/firestore";
+import { router } from "expo-router";
+import { WarningCircle } from "phosphor-react-native";
+import { useState } from "react";
+import { useForm, Controller } from "react-hook-form";
+import {
+  View,
+  Text,
+  TouchableWithoutFeedback,
+  KeyboardAvoidingView,
+  Keyboard,
+  TextInput,
+  Pressable,
+} from "react-native";
+import * as yup from "yup";
+import Spinner from "@/components/Spinner";
+
+const schema = yup.object({
+  emailOrUsername: yup.string().required("Email or username is required"),
+});
+
+type ForgotPasswordFormData = {
+  emailOrUsername: string;
+};
+
+const ForgotPassword = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const auth = getAuth();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isValid },
+    setError,
+  } = useForm<ForgotPasswordFormData>({
+    resolver: yupResolver(schema),
+    mode: "onSubmit",
+    defaultValues: {
+      emailOrUsername: "",
+    },
+  });
+
+  const findEmail = async (data: string): Promise<string> => {
+    const userSnapshot = await firestore()
+      .collection("users")
+      .where("username", "==", data)
+      .limit(1)
+      .get();
+
+    if (!userSnapshot.empty) {
+      const userData = userSnapshot.docs[0].data();
+      return userData.email;
+    }
+
+    return data;
+  };
+
+  const onSubmit = async (data: ForgotPasswordFormData) => {
+    setIsLoading(true);
+
+    try {
+      const identifier = data.emailOrUsername.trim();
+
+      const emailToUse = await findEmail(identifier);
+
+      await sendPasswordResetEmail(auth, emailToUse);
+
+      router.push({
+        pathname: "/(auth)/confirmReset",
+        params: { email: emailToUse },
+      });
+    } catch (error) {
+      const firebaseError = error as { code: string };
+
+      console.error("Forgot password error:", error);
+      if (firebaseError.code === "auth/invalid-email") {
+        setError("emailOrUsername", {
+          message: "Please enter a valid email address.",
+        });
+      } else if (firebaseError.code === "auth/user-not-found") {
+        setError("emailOrUsername", {
+          message: "No user found with this email address.",
+        });
+      }
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <View className="flex-1 bg-gray-950">
+      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+        <KeyboardAvoidingView
+          behavior="padding"
+          className="my-16 flex-1 flex-col px-6 py-4"
+        >
+          <Text className="pbk-h5 mb-4 text-base-white">Forgot password?</Text>
+
+          <Text className="pbk-b1 mb-8 text-base-white">
+            Enter your email address or username and we will send you a link to
+            reset your password.
+          </Text>
+
+          <Text className="pbk-b2 mb-1.5 text-base-white">
+            Email or username
+          </Text>
+          <Controller
```

Comment:
Nitpick, but the spacing between this form input and the next button isn't consistent with the other login screens

Reply:
Thanks for noticing this, I added a View component around the conditional error that acts as a spacer similar to what I have in the login screen. It needs to be a separate View because I can't have it under the conditional error as that would only space it appropriately when the error shows, and I also can't have it under the Controller because that would add too much space between the error message and the input field

```suggestion
            <View className="mb-5">
            {errors.emailOrUsername && (
              <Text className="pbk-b3 mb-4 text-red-600">
                {errors.emailOrUsername.message}
              </Text>
            )}
          </View>
```